### PR TITLE
MEL4-295 adding client package

### PR DESCRIPTION
**Pull request by shaulzorea**

### CODE REVIEW

File: packages/client/src/generated-clients/http-client-typescript/.openapi-generator/FILES [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.openapi-generator-ignore
```
Comment (shaikegross) [inline packages/client/src/generated-clients/http-client-typescript/.openapi-generator/FILES:1]: Whats this file and the VERISON file?

File: packages/client/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "Typescript client for Vendor Directory API",
+  "repository": "https://github.com/melio/vendor-directory-api",
+  "publishConfig": {
+    "registry": "https://npm.pkg.github.com/melio"
+  },
+  "scripts": {
+    "build": "tsc",
+    "generate:models": "openapi-generator generate  -i ../vendor-directory-api/openapi.yaml -g typescript-fetch -o src/generated-clients/http-client-typescript --additional-properties=typescriptThreePlus=true",
+    "prepare": "npm run build"
+    "publish": "npm publish",
```
Comment (shaikegross) [inline packages/client/package.json:13]: no need to add publish script. publishing works out of the box with NPM when running `npm publish`

File: packages/client/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    /* Basic Options */
+    // "incremental": true,                         /* Enable incremental compilation */
+    "target": "ESNext",                                /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
+    "module": "commonjs",                           /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
+    // "lib": [],                                   /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                             /* Allow javascript files to be compiled. */
+    // "checkJs": true,                             /* Report errors in .js files. */
+    // "jsx": "preserve",                           /* Specify JSX code generation: 'preserve', 'react-native', 'react', 'react-jsx' or 'react-jsxdev'. */
+    // "declaration": true,                         /* Generates corresponding '.d.ts' file. */
```
Comment (shaikegross) [inline packages/client/tsconfig.json:13]: I think this needs to be uncommented so it can be imported in a ts module

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,5 @@
+.DS_Store
```
Comment (shaikegross) [inline .gitignore:1]: Add a title? I like when gitignore has titles for what was ignored. like: 

`# mac`
`.DS_Store`

**Review by shaikegross — CHANGES_REQUESTED** [review]
Hey please uncomment the declaration setting in tsconfig and remove the publish script.

There are some other comments but they aren't "deal breakers"

File: .gitignore (hunk 4 above)
Comment (shaulzorea) [reply .gitignore:2]: 👍🏽

File: packages/client/tsconfig.json (hunk 3 above)
Comment (shaulzorea) [reply packages/client/tsconfig.json]: 👍🏽

File: packages/client/package.json (hunk 2 above)
Comment (shaulzorea) [reply packages/client/package.json]: 👍🏽

File: packages/client/src/generated-clients/http-client-typescript/.openapi-generator/FILES (hunk 1 above)
Comment (shaikegross) [reply packages/client/src/generated-clients/http-client-typescript/.openapi-generator/FILES]: hmmm no need to gitignore. my bad

**Review by shaikegross — APPROVED** [review]

**Pull request merged**